src/Unused/Regex.hs

```diff
@@ -0,0 +1,15 @@
+{-# LANGUAGE FlexibleContexts #-}
+
+module Unused.Regex
+    ( matchRegex
+    ) where
+
+import Text.Regex.TDFA
+
+matchRegex :: String -> String -> Bool
+matchRegex r =
+    matchTest $ mkRegex r
+
+mkRegex :: RegexMaker Regex CompOption ExecOption String => String -> Regex
```

Comment:
@jferris I'd found this on GitHub I think. I'd had similar thoughts a while back, here's the compiler error:

```
/Users/joshuaclayton/dev/personal/unused/src/Unused/Regex.hs:10:14:
    No instance for (RegexLike regex0 String)
      arising from a use of ‘matchTest’
    The type variable ‘regex0’ is ambiguous
    Note: there is a potential instance available:
      instance RegexLike Regex String
        -- Defined in ‘Text.Regex.TDFA.String’
    In the first argument of ‘(.)’, namely ‘matchTest’
    In the expression: matchTest . makeRegex
    In an equation for ‘matchRegex’: matchRegex = matchTest . makeRegex

/Users/joshuaclayton/dev/personal/unused/src/Unused/Regex.hs:10:26:
    No instance for (RegexMaker regex0 compOpt0 execOpt0 String)
      arising from a use of ‘makeRegex’
    The type variables ‘regex0’, ‘compOpt0’, ‘execOpt0’ are ambiguous
    Note: there is a potential instance available:
      instance RegexMaker Regex CompOption ExecOption String
        -- Defined in ‘Text.Regex.TDFA.String’
    In the second argument of ‘(.)’, namely ‘makeRegex’
    In the expression: matchTest . makeRegex
    In an equation for ‘matchRegex’: matchRegex = matchTest . makeRegex
```

After playing around with this, I did get this to work:

```haskell
{-# LANGUAGE FlexibleContexts #-}

module Unused.Regex
    ( matchRegex
    ) where

import Text.Regex.TDFA
import qualified Text.Regex.TDFA.String as R

matchRegex :: String -> String -> Bool
matchRegex = matchTest . (makeRegex :: String -> R.Regex)
```

Buuuut it seems a little strange. Any other thoughts here?